js_src/dismiss.ts

```diff
@@ -0,0 +1,40 @@
+export default class Dismiss {
+    private refersToElement;
+    private doesSoftDismiss;
+
+    public static softDismissDuration = 250; // ms
+
+    constructor(public element) {
+        this.refersToElement = document.querySelector(element.getAttribute("data-dismiss"));
+
+        this.doesSoftDismiss = element.hasAttribute("data-dismiss-soft");
+        
+        element.addEventListener("click", () => { this.dismiss() });
+    }
+
+    private dismiss() {
+        // Remove immediately, if no support for animate
+        if (!this.doesSoftDismiss || !this.refersToElement.animate) {
```

Comment:
as mentioned above, `this.refersToElement.animate` would cause a typeError if `this.refersToElement` was `null`